app/models/goal.py

```diff
@@ -3,4 +3,21 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True,autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+
+
+    def to_dict(self):
+        return {
+            "id": self.goal_id,
+            "title": self.title,
+        }
+
+    def to_dict_with_tasks(self):
+        return {
+            "id": self.goal_id,
+            "title": self.title,
+            "tasks": [task.to_dict() for task in self.tasks]
```

Comment:
Consider how you could DRY up your code with a single method that include "tasks" as a key based on the value of an argument, for instance